# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kvm/16-bios.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

# Send to upstream BOCHS
This ACPI BIOS patch provides:
- The PCI bus definition for PIC, HPET, RTC, SMC and OSYS and SMIF
- An PBLK which provides the size of the L2 and L3 caches
- If newer hardware like ICH6 IDE is found, init it
Signed-off-by: Alex Graf - http://alex.csgraf.de
Index: qemu-kvm-0.12.1.2/roms/seabios-mac/src/acpi-dsdt.dsl
===================================================================
--- qemu-kvm-0.12.1.2.orig/roms/seabios-mac/src/acpi-dsdt.dsl
+++ qemu-kvm-0.12.1.2/roms/seabios-mac/src/acpi-dsdt.dsl
@@ -194,6 +194,7 @@ DefinitionBlock (
 
         Device(HPET) {
             Name(_HID,  EISAID("PNP0103"))
+            Name(_CID, 0x010CD041)
             Name(_UID, 0)
             Method (_STA, 0, NotSerialized) {
                     Return(0x0F)
@@ -210,6 +211,23 @@ DefinitionBlock (
                 )
             })
         }
+        Device (SMC)
+        {
+            Name (_HID, EisaId ("APP0001"))
+            Name (_CID, "smc-napa")
+            Name (_STA, 0x0B)
+            Name (_CRS, ResourceTemplate ()
+            {
+                IO (Decode16,
+                    0x0300,             // Range Minimum
+                    0x0300,             // Range Maximum
+                    0x01,               // Alignment
+                    0x20,               // Length
+                    )
+                IRQNoFlags ()
+                    {6}
+            })
+        }
     }
 
     Scope(\_SB.PCI0) {
Index: qemu-kvm-0.12.1.2/roms/seabios-mac/src/pciinit.c
===================================================================
--- qemu-kvm-0.12.1.2.orig/roms/seabios-mac/src/pciinit.c
+++ qemu-kvm-0.12.1.2/roms/seabios-mac/src/pciinit.c
@@ -19,7 +19,8 @@ static u32 pci_bios_mem_addr;
 static u32 pci_bios_bigmem_addr;
 /* host irqs corresponding to PCI irqs A-D */
 static u8 pci_irqs[4] = {
-    10, 10, 11, 11
+//    10, 10, 11, 11
+    11, 10, 11, 10
 };
 
 static void pci_set_io_region_addr(u16 bdf, int region_num, u32 addr)
@@ -54,7 +55,9 @@ static void pci_bios_init_bridges(u16 bd
 
     if (vendor_id == PCI_VENDOR_ID_INTEL
         && (device_id == PCI_DEVICE_ID_INTEL_82371SB_0
-            || device_id == PCI_DEVICE_ID_INTEL_82371AB_0)) {
+            || device_id == PCI_DEVICE_ID_INTEL_82371AB_0
+            || device_id == PCI_DEVICE_ID_INTEL_82945GM_HB
+            || device_id == PCI_DEVICE_ID_INTEL_ICH7_1)) {
         int i, irq;
         u8 elcr[2];
 
@@ -91,7 +94,8 @@ static void pci_bios_init_device(u16 bdf
     case PCI_CLASS_STORAGE_IDE:
         if (vendor_id == PCI_VENDOR_ID_INTEL
             && (device_id == PCI_DEVICE_ID_INTEL_82371SB_1
-                || device_id == PCI_DEVICE_ID_INTEL_82371AB)) {
+                || device_id == PCI_DEVICE_ID_INTEL_82371AB
+                || device_id == PCI_DEVICE_ID_INTEL_ESB2_18)) {
             /* PIIX3/PIIX4 IDE */
             pci_config_writew(bdf, 0x40, 0x8000); // enable IDE0
             pci_config_writew(bdf, 0x42, 0x8000); // enable IDE1
